### PR TITLE
Smart Wallets + Auth: Enable All Smart Wallet Chains For `CrossmintAuthProvider` & `CrossmintWalletProvider`

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

We temporarily only had staging chains enabled for these providers, this PR updates the SDK to enable the use of all Smart Wallet supported chains.

## Test plan

Sanity checked that the auth + smart wallet demo flow still works.

## Package updates

This PR patches both `@crossmint/client-sdk-smart-wallet` and `@crossmint/client-sdk-react-ui`

### CODE REVIEW

**Review by jmderby — APPROVED** [review]
lgtm

File: packages/client/wallets/smart-wallet/src/index.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ export type {
     WalletParams,
 } from "./types/params";
 
+export type { SmartWalletChain } from "./blockchain/chains";
```
Comment (alfonso-paella) [inline packages/client/wallets/smart-wallet/src/index.ts:14]: I wonder if this type should be EVMSmartWalletChain. I think in the future we may benefit from disallowing one to pass "type": "solana-smart-wallet", "defaultChain": "polygon-amoy"

File: packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import { useCrossmint } from "@/hooks";
 import { ReactNode, createContext, useEffect, useMemo, useState } from "react";
 
-import { EVMSmartWallet, SmartWalletError, SmartWalletSDK } from "@crossmint/client-sdk-smart-wallet";
+import { EVMSmartWallet, SmartWalletChain, SmartWalletError, SmartWalletSDK } from "@crossmint/client-sdk-smart-wallet";
 
 export type CrossmintWalletConfig = {
     type: "evm-smart-wallet";
-    defaultChain: "polygon-amoy" | "base-sepolia" | "optimism-sepolia" | "arbitrum-sepolia";
+    defaultChain: SmartWalletChain;
```
Comment (alfonso-paella) [inline packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx:8]: Do we have a good error if you pass a chain that's not compatible with the environment your API key is set to?

File: packages/client/wallets/smart-wallet/src/index.ts (hunk 1 above)
Comment (devlyn37) [reply packages/client/wallets/smart-wallet/src/index.ts]: makes sense, renamed.

File: packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx (hunk 2 above)
Comment (devlyn37) [reply packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx]: No, mind if I add that as a follow up?

Comment (devlyn37) [reply packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx]: will do so unless you object!

**Pull request merged**